src/Makefile

```diff
@@ -16,6 +16,8 @@ endif
 
 CFLAGS ?= -g -O2 -Werror -Wall
 ALL_CFLAGS += $(CFLAGS)
+ALL_CFLAGS += $(shell command -v pkg-config > /dev/null 2>&1 && pkg-config --cflags libelf)
+LDFLAGS += $(shell command -v pkg-config > /dev/null 2>&1 && pkg-config --libs libelf || echo "-lelf")
```

Comment:
Let's handle `LDFLAGS` similar to `CFLAGS`:
* As recommended in [1] for similar case, add `ALL_LDFLAGS` and set your option there, `LDFLAGS` should be fully overridable;
* Move this under `ifndef NO_PKG_CONFIG` -- together with `ALL_CLFAGS`.
* Use `$(PKG_CONFIG)` w/o fallback under `ifndef NO_PKG_CONFIG`, but add `-lelf` otherwise.

Together with previous line it can be something like:
```
...
ALL_LDFLAGS += $(LDFLAGS)
ifdef NO_PKG_CONFIG
        ALL_LDFLAGS += -lelf
else
        ALL_CFLAGS += $(shell $(PKG_CONFIG) --cflags libelf)
        ALL_LDFLAGS += $(shell $(PKG_CONFIG) --libs libelf)
endif
```

`NO_PKG_CONFIG` should also be documented in the README along your other changes.

[1] https://www.gnu.org/software/make/manual/html_node/Command-Variables.html#Command-Variables

Reply:
Done as well, thanks!